google/generativeai/retriever.py

```diff
@@ -30,14 +30,11 @@
 from google.generativeai.types import safety_types
 from google.generativeai.types.model_types import idecode_time
 
-_CORPORA_NAME_REGEX = re.compile(r"^corpora/[a-z0-9-]+")
-_REMOVE = string.punctuation
-_REMOVE = _REMOVE.replace("-", "")  # Don't remove hyphens
-_PATTERN = r"[{}]".format(_REMOVE)  # Create the pattern
+_VALID_NAME = r"^[^-][a-zA-Z\d-][^-]+$"
 
 
 def create_corpus(
-    name: Optional[str] = None,
+    name: str,
     display_name: Optional[str] = None,
```

Comment:
Does it matter that `display_name` here is Optional, but on the Corpus dataclass, it's non-optional?

Reply:
Updated all `display_name`s to be optional -- probably an oversight of mine when i was updating these